### PR TITLE
Start cleaning up the USB & Ja Rule code now that it's working.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
More cleanup.

File: libs/LICENCE [hunk 1]
```diff
@@ -0,0 +1,13 @@
+This program is free software; you can redistribute it and/or modify
```
Comment (peternewman) [inline libs/LICENCE]: Should this not be "This library..."?

File: libs/usb/Types.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Types.h
+ * Types use with the USB subsystem.
```
Comment (peternewman) [inline libs/usb/Types.h]: SPaG used

File: libs/usb/Types.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+#ifndef LIBS_USB_TYPES_H_
+#define LIBS_USB_TYPES_H_
+
+#include <utility>
+
+namespace ola {
+namespace usb {
+
+/**
+ * @brief Represents a device on the USB.
```
Comment (peternewman) [inline libs/usb/Types.h]: This is another PAT Tester isn't it, but it still reads oddly. How about "Represents a USB device"?

File: plugins/usbdmx/AsyncPluginImpl.cpp [hunk 4]
```diff
@@ -290,66 +321,63 @@ bool AsyncPluginImpl::USBDeviceAdded(libusb_device *usb_device) {
...
-      &m_device_factory_map, usb_device);
-  if (factory) {
-    factory->DeviceRemoved(&m_widget_observer, usb_device);
-  }
 }
 
 /*
  * @brief Signal widget / device addition.
- * @param widget The widget that was added.
+ * @param device_id The device id to add.
```
Comment (peternewman) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: Is it worth saying USBDeviceID here to differentiate from OLA device IDs?

File: plugins/usbdmx/AsyncPluginImpl.cpp [hunk 5]
```diff
@@ -290,66 +321,63 @@ bool AsyncPluginImpl::USBDeviceAdded(libusb_device *usb_device) {
...
   }
   m_plugin_adaptor->RegisterDevice(device);
+  state->ola_device = device;
   return true;
 }
 
 /*
  * @brief Signal widget removal.
- * @param widget The widget that was removed.
+ * @param device_id The device id to remove.
```
Comment (peternewman) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: Again

**Comment by peternewman** [comment]
Generally LGTM

File: libs/LICENCE (hunk 1 above)
Comment (nomis52) [inline libs/LICENCE]: These bits are 'ola internal' so still under the GPL (no change from plugins/)

File: libs/usb/Types.h (hunk 2 above)
Comment (nomis52) [inline libs/usb/Types.h]: Fixed

File: libs/usb/Types.h (hunk 3 above)
Comment (nomis52) [inline libs/usb/Types.h]: Yup "Universal Serial Bus Bus" 

Changed

File: plugins/usbdmx/AsyncPluginImpl.cpp (hunk 4 above)
Comment (nomis52) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: The type appears in the doxygen anyway. People reading the code can see the type in the signature.

File: libs/LICENCE (hunk 1 above)
Comment (peternewman) [inline libs/LICENCE]: Ah okay

File: plugins/usbdmx/AsyncPluginImpl.cpp (hunk 4 above)
Comment (peternewman) [inline plugins/usbdmx/AsyncPluginImpl.cpp]: Okay

**Comment by peternewman** [comment]
LGTM

**Comment by peternewman** [comment]
LGTM

**Pull request merged**